docs/deployment/pull-backup.rst

```diff
@@ -174,3 +188,116 @@ directly extract it without the need of mounting with SSHFS:
+standard input and output available to a unix socket::
+
+   borg-server:~$ socat UNIX-LISTEN:/run/borg/reponame.sock,fork EXEC:"borg serve --append-only --restrict-to-path /path/to/repo"
+
+Socat will wait until a connection is opened. Then socat will execute the
+command given, redirecting Standard Input and Output to the unix socket. The
+optional arguments for ``borg serve`` are not necessary but a sane default.
+
+.. note::
+   When used in production you may also use systemd socket-based activation
+   instead of socat on the server side. You would wrap the ``borg serve`` command
+   in a `service unit`_ and configure a matching `socket unit`_
+   to start the service whenever a client connects to the socket.
+
+   .. _service unit: https://www.freedesktop.org/software/systemd/man/systemd.service.html
+   .. _socket unit: https://www.freedesktop.org/software/systemd/man/systemd.socket.html
+
+Now we need a way to access the unix socket on *borg-client* (holding the
+data to be backed up), as we created the unix socket on *borg-server*
+Opening a SSH connection from the *borg-server* to the *borg-client* with reverse port
+forwarding can do this for us::
+
+   borg-server:~$ ssh -R /run/borg/reponame.sock:/run/borg/reponame.sock borgc@borg-client
+
+.. note::
+
+   As the default value of OpenSSH for ``StreamLocalBindUnlink`` is ``no``, the
+   socket file created by sshd is not removed. Trying to connect a second time,
+   will print a short warning, and the forwarding does **not** take place::
+
+      Warning: remote port forwarding failed for listen path /run/borg/reponame.sock
+
+   When you are done, you have to manually remove the socket file, otherwise
+   you may see an error like this when trying to execute borg commands::
+
+      Remote: YYYY/MM/DD HH:MM:SS socat[XXX] E connect(5, AF=1 "/run/borg/reponame.sock", 13): Connection refused
+      Connection closed by remote host. Is borg working on the server?
+
+
+When a process opens the socket on *borg-client*, SSH will forward all
+data to the socket on *borg-server*.
+
+The next step is to tell borg on *borg-client* to use the unix socket to communicate with the
+``borg serve`` command on *borg-server* via the socat socket instead of SSH::
+
+   borg-client:~$ export BORG_RSH="sh -c 'exec socat STDIO UNIX-CONNECT:/run/borg/reponame.sock'"
+
+The default value for ``BORG_RSH`` is ``ssh``. By default Borg uses SSH to create
+the connection to the backup server. Therefore Borg parses the repo URL
+and adds the server name (and other arguments) to the SSH command. Those
+arguments can not be handled by socat. We wrap the command with ``sh`` to
+ignore all arguments intended for the SSH command.
+
+All Borg commands can now be executed on *borg-client*. For example to create a
+backup execute the ``borg create`` command::
+
+   borg-client:~$ borg create ssh://borg-server/path/to/repo::archive /path_to_backup
+
+When creating a backup should be scheduled or otherwise automated, the
```

Comment:
While I'm not a native speaker myself, I have a feeling that a native speaker would not put it this way. Had to reread the first part before I got what it is trying to say, too.
How about simply:
When automating backup creation, the ...

Reply:
-> edb463f9